dashboard/app/models/pd/enrollment.rb

```diff
@@ -182,7 +182,7 @@ def email=(value)
   end
 
   def resolve_user
-    user || User.find_by_email_or_hashed_email(email)
+    user || User.find_by_email_or_hashed_email(email) || User.all.find {|u| u.email_for_enrollments == email}
```

Comment:
apologies, I just noticed that `User.all` is going to be prohibitively expensive! in production, this would query the database for all 100M users and try to load them into memory, which could be a problem both for the load on the database and also the memory usage on the server.

can you voice over why the change on this line is important? if it is just part of `Verify that the existing enrollment.email field is already being used in all the places we would want an “alternate email” field to be used`, then it may be possible to skip.

Reply:
This is to be able to still find the user based on the enrollment's email in the case of the enrollment's email being the alternate email (which doesn't have a column in the user db, it just exists as part of the user's application data). I agree it seems very expensive, do you think there is a workaround?

Reply:
Yes, we can come up with a workaround, but it may depend on the user scenario we are trying to fix exactly how (and whether) to work around it. we should review the user scenarios to make sure which ones we are trying to fix before proceeding.

can you find all of the callers of `.resolve_user`, identify the user scenario, and say whether/why it is important to look up the user by alternate email in that case? if you have any tests that fail when you revert the change on this line, that could be a shortcut to answering the question.

Reply:
I believe I've found a work around using the user's application. In this case, `enrollment.rb` already has access to the user's application so that was a quick fix.

Reply:
Nice find, Turner! Let's use our scheduled 1:1 time tomorrow to discuss this PR. I am still not clear on why this check needs to be modified, based on the context in which it is being called.

Reply:
per offline discussion, Turner is going to look into passing the user id through the create enrollment form, so that we do not need to rely on the alternate email to locate the user.